src/set_theory/principal.lean

```diff
@@ -67,4 +69,37 @@ theorem nfp_le_of_principal {op : ordinal → ordinal → ordinal}
   {a o : ordinal} (hao : a < o) (ho : principal op o) : nfp (op a) a ≤ o :=
 nfp_le.2 $ λ n, le_of_lt (ho.iterate_lt hao n)
 
+/-! ### Principal ordinals are unbounded -/
+
+/-- The least strict upper bound of `op` applied to all pairs of ordinals less than `o`. -/
+def blsub₂ (op : ordinal → ordinal → ordinal) (o : ordinal) : ordinal :=
```

Comment:
Is there really no clearer name? If not, can you explain the naming convention in one of the module docstrings?

Reply:
I don't find the name particularly unclear. After all, this is pretty much just a two-argument version of `blsub`. I added a short comment clarifying this.

Reply:
I have no idea what `blsup` stands for 😅 
This name is fine given the rest, but it's not fine on its own, so I'm just suggesting you explain all the names once and for all somewhere.

Reply:
`blsub` stands for bounded (as in, made from a family `\Pi a < o, ordinal`) least strict upper bound. That name is explained in `ordinal_arithmetic`, where it first appears, and is consistent with the nomenclature throughout the file.

Reply:
Okay sure